mlflow/utils/rest_utils.py

```diff
@@ -93,13 +140,17 @@ def http_request(
     )
 
     timeout = MLFLOW_HTTP_REQUEST_TIMEOUT.get() if timeout is None else timeout
-    hostname = host_creds.host
     auth_str = None
     if host_creds.username and host_creds.password:
         basic_auth_str = f"{host_creds.username}:{host_creds.password}".encode()
         auth_str = "Basic " + base64.standard_b64encode(basic_auth_str).decode("utf-8")
     elif host_creds.token:
         auth_str = f"Bearer {host_creds.token}"
+    elif host_creds.client_secret:
+        raise MlflowException(
+            "To use OAuth authentication, set environmental variable "
+            f"'{MLFLOW_ENABLE_DB_SDK.name}' to true"
```

Comment:
Can we set some error code to `MlflowException`?

Reply:
Which error code do you prefer ? By default it is set with `INTERNAL_ERROR` code

Reply:
Probably `401`? `500` conventionally indicates server-side error that users don't need to take action.